Ldap/LdapManager.php

```diff
@@ -194,4 +207,60 @@ public static function unescapeValue($values = array())
 
         return (count($values) == 1 && array_key_exists(0, $values)) ? $values[0] : $values;
     }
+
+    /**
+     * Add roles based on role configuration from user memberOf attribute.
+     *
+     * @param UserInterface
+     * @param array $entry
+     */
+    private function addRolesFromMemberof($user, $entry)
+    {
+        if (isset($entry['memberof'])) {
+            foreach ($entry['memberof'] as $role) {
+                $dnSuffixFilter = $this->params['role']['memberOf']['dnSuffixFilter'];
+
+                if (preg_match("/^cn=(.*),$dnSuffixFilter/", $role, $roleName)) {
+                    $user->addRole(sprintf('ROLE_%s',
```

Comment:
I agree that this should be moved to the user class but what do you mean exactly with a trait.
Something like described here: http://php.net/manual/de/language.oop5.traits.php
I've never seen that in "field" before or used it. Wouldn't that increase the complexity?